js/CRM_Custom_Form_Field.js

```diff
@@ -22,16 +34,33 @@ CRM.$(function($) {
       $('#autoinc-min-value').show();
       $('#hideDefault, #hideDesc, .crm-custom-field-form-block-is_required, .crm-custom-field-form-block-is_view').hide();
 
+      // If civicrm v5.32.2
+      if ($('#html_type').length) {
+        $(htmlType).attr('disabled', 'disabled');
+      }
+
       // Add Autoincrement text next to Integer if autoinc field is checked
       var $dataTypeField = $('.crm-custom-field-form-block-data_type .crm-frozen-field');
       $dataTypeField.html($dataTypeField.html().replace('Integer', ts('Integer (Autoincrement)')));
     }
 
     // Show autoincfield if data type matches
-    if($('#data_type_0').val() == 1 && $('#data_type_1').val() === 'Text') {
+    if($(dataType).val() == 1 && $(htmlType).val() === 'Text') {
       $('#autoinc-field').show();
     }
 
+    // Match data_type and html_type
+    $(document).on('change', dataType + ', ' + htmlType, function(){
```

Comment:
Because these two DOM elements already exist (correct?), I think we could use `$(dataType + ', ' + htmlType).change(function(){` here (applying the change handler to the elements themselves, rather than to the document); that's basically what was there before, except it uses variables instead of strings for the jquery selector. I think that would be slightly easier to read for other devs. Could you try that?

Reply:
I need to change and use the $(document) for this line since in civicrm 5.32.2, the element is loaded via ajax. If I only use .change and not .on('change') the event will not trigger.

Reply:
Got it, thanks.